### DISCUSSION TITLE
SKLearn influence on loss?

### DISCUSSION THREAD

**Discussion by ChristianZX** [alan-turing-institute/sktime#2624] [Q&A]
Hi everybody,

I’ve recently started using SKLearn and mostly used convinience features.
Does Sklearn also have an influnece on the loss and are there compelling examples illustrating that?
I failed to find any myself.

**Comment by fkiraly** [comment]
Hello @ChristianZX, can you kindly explain what you mean by "influence on the loss"?

Do you mean performance metrics, which include loss functions?

If so, then note that for forecasting we went in a different design direction, since loss functions for forecasting and their application tend to be more complex, e.g., you can have baseline values or train values. They are also more often parametric.
Hence, the `sktime` approach is making metrics estimator-like objects rather than functions.

For classification, the type of metrics and the metrics design are identical, and you can use the `sklearn` ones directly, of course.

**Reply by ChristianZX** [reply]
Hi @fkiraly. Thanks for the answer!
What I meant with the question was this: 
Are there examples where SKlearn has a lower mean squared error, let's say using a random forrest (or whatever else) while compared to a Sklearn Random Forrest using the same training data.

Hope I phrased it more precisely this time. :)

**Reply by fkiraly** [reply]
sorry, I'm not sure I understand. You are saying you want to compare an sklearn random forest to an sklearn random forest - so the same thing with itself?

That's probably not what you mean?

Did you mean sktime random forest with sklearn random forest?

If so, the first thing to note is the difference of categories. sklearn does supervised classification etc, and sktime does time series classification or forecasting etc. These are different learning tasks, and sktime provides an sklearn-like interface to the different learning tasks.

So, direct comparisons are not possible conceptually, since the algorithms are for different tasks.
Of course, you can combine an sklearn random forest with tabularization, or with direct forecasting reduction, but then the point of comparison would not be clear (what is the second thing you want to compare to?).
